openraft/src/core/install_snapshot.rs

```diff
@@ -34,28 +33,16 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> RaftCore<C,
     ) -> Result<InstallSnapshotResponse<C::NodeId>, InstallSnapshotError<C::NodeId>> {
         tracing::debug!(req = display(req.summary()));
 
-        if req.vote < self.engine.state.vote {
-            tracing::debug!(?self.engine.state.vote, %req.vote, "InstallSnapshot RPC term is less than current term");
-
+        if self.engine.handle_vote_change(&req.vote).is_err() {
+            tracing::info!(?self.engine.state.vote, %req.vote, "InstallSnapshot RPC term is less than current term, ignoring it.");
             return Ok(InstallSnapshotResponse {
                 vote: self.engine.state.vote,
             });
         }
+        self.run_engine_commands::<Entry<C>>(&[]).await?;
```

Comment:
No matter if `handle_vote_change()` returns error or not, the output commands should be consumed. There will still be some commands to execute even when it returns error.

The return statement goes after `self.run_engine_commands::<Entry<C>>(&[]).await?;`